cost-model/src/cost_model.rs

```diff
@@ -228,13 +249,36 @@ impl CostModel {
         )
     }
 
+    /// Return (programs_execution_cost, loaded_accounts_data_size_cost)
+    fn get_estimated_execution_cost(
+        transaction: &impl SVMMessage,
+        feature_set: &FeatureSet,
+    ) -> (u64, u64) {
+        // if failed to process compute_budget instructions, the transaction will not be executed
+        // by `bank`, therefore it should be considered as no execution cost by cost model.
+        let (programs_execution_costs, loaded_accounts_data_size_cost) =
+            match process_compute_budget_instructions(
+                transaction.program_instructions_iter(),
+                feature_set,
+            ) {
+                Ok(compute_budget_limits) => (
+                    u64::from(compute_budget_limits.compute_unit_limit),
+                    Self::calculate_loaded_accounts_data_size_cost(
+                        compute_budget_limits.loaded_accounts_bytes.get(),
+                        feature_set,
+                    ),
+                ),
+                Err(_) => (0, 0),
+            };
+
+        (programs_execution_costs, loaded_accounts_data_size_cost)
+    }
+
     /// Return the instruction data bytes cost.
-    fn get_instructions_data_cost(transaction: &SanitizedTransaction) -> u64 {
+    fn get_instructions_data_cost(transaction: &impl SVMMessage) -> u64 {
         let ix_data_bytes_len_total: u64 = transaction
-            .message()
-            .instructions()
-            .iter()
-            .map(|instruction| instruction.data.len() as u64)
+            .program_instructions_iter()
```

Comment:
nit: we can get by with using `instruction_iter` and avoid the extra work of getting the program id in this iterator

Reply:
4472e56f529364144dea0b320a7935487e2fe623 for it